packages/website/bin/page-generator/write-pages.js

```diff
@@ -0,0 +1,230 @@
+const fse = require('fs-extra');
+const path = require('path');
+
+const {
+  changelogTemplate,
+  exampleTemplate,
+  singleComponentTemplate,
+  wrappedComponentTemplate,
+} = require('./templates');
+
+const writeFile = (pagePath, content) => {
+  fse.ensureFileSync(pagePath);
+  fse.writeFileSync(pagePath, content);
+};
+
+/**
+ * Generates an import path to a destination file relative from a given path
+ * @param from absolute path of the js file doing the import
+ * @param to absolute path of the file being imported
+ */
+const getImportPath = (from, to) => {
+  const fromDir = path.dirname(from);
+  if (!to) {
+    throw new Error('To path cannot be empty');
+  }
+  return path.relative(fromDir, to).replace('.js', '');
+};
+
+const getGenericPageInfo = (
+  pagesPath,
+  pagePath,
+  componentPath,
+  wrappersPath,
+  wrapperName,
+) => {
+  const absolutePagePath = path.resolve(pagesPath, pagePath);
+  const componentImportPath = componentPath
+    ? getImportPath(absolutePagePath, componentPath)
+    : undefined;
+  const packageHomeWrapperPath = getImportPath(
+    absolutePagePath,
+    path.join(wrappersPath, `${wrapperName}.js`),
+  );
+
+  return {
+    absolutePagePath,
+    componentImportPath,
+    packageHomeWrapperPath,
+  };
+};
+
+const generateBasicPage = (
+  pagePath,
+  componentPath,
+  data,
+  wrapperName,
+  { wrappersPath, pagesPath },
+  title = '',
```

Comment:
What's the work to add this as part of data before we get here?

Reply:
the larger piece of work is to change the function signatures of everything in this file to be more consistent and easier to use. one part of that is removing the title arg and having consumers pass pageTitle in as a property of data